UML-Diagram/SearchSystem.puml

```diff
@@ -15,13 +14,15 @@ package "Search System" {
     }
 
     abstract class PatronSearch extends SearchBase {
+        - PatronNotFoundException : PatronNotFoundException
         - PatronManager Patrons
         + override DisplayDetails()
         + override NotResultFound()
         + abstract Search()
     }
 
     abstract class BookSearch extends SearchBase {
+        - BookNotFoundException : BookNotFoundException
```

Comment:
I dont think that its needed in the constructor of the classes maybe could be implemented in the methods, but has an atribute in the class i dont think it should be, Or you refere to an instance that should be initialized on  the atributes but not included on constructor of the class.

Reply:
Actually, I was waiting for that question. I only added to generate like a relationship of association between them due to when it's going to be used as throw new ExceptionType(...)

Reply:
ho in that case you can use the relation of use like BorrowManager ..> ExceptionHandler for example